styx-scheduler-service/src/main/java/com/spotify/styx/cleaner/CleanerOperation.java

```diff
@@ -0,0 +1,31 @@
+/*-
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.cleaner;
+
+public interface CleanerOperation {
```

Comment:
Is this conceptually different from the pattern of using Closeable and Closer?

Reply:
Or, I guess this is meant to get invoked repeatedly... Then why is this not just a Runnable? If there's some design thinking about what is going to be  specific in CleanerOperation's used and evolution, then documenting it seems like a good idea.

Reply:
Sure we can just use `RunnableWithException` in `styx-common`.

Reply:
It might still be a good idea to have this specialised interface than the very generic `RunnableWithException`.